webui/react/src/pages/F_ExpList/glide-table/custom-renderers/utils.ts

```diff
@@ -35,3 +35,65 @@ export function roundedRect(
   ctx.arcTo(x, y + height, x, y + height - radius.bl, radius.bl);
   ctx.arcTo(x, y, x + radius.tl, y, radius.tl);
 }
+
+export function drawArrow(
+  ctx: CanvasRenderingContext2D,
+  direction: 'down' | 'up' = 'up',
+  x: number,
+  y: number,
+  width = 8,
+  height = 12,
+): void {
+  const headDelta = width / 2;
+
+  ctx.beginPath();
+
+  if (direction === 'up') {
+    ctx.moveTo(x, y + headDelta);
+    ctx.lineTo(x + headDelta, y);
+    ctx.lineTo(x + width, y + headDelta);
+    ctx.moveTo(x + headDelta, y);
+    ctx.lineTo(x + headDelta, y + height);
+  } else {
+    ctx.moveTo(x, y + height - headDelta);
+    ctx.lineTo(x + headDelta, y + height);
+    ctx.lineTo(x + width, y + height - headDelta);
+    ctx.moveTo(x + headDelta, y);
+    ctx.lineTo(x + headDelta, y + height);
+  }
+
+  ctx.closePath();
+  ctx.stroke();
+}
+
+function textWithEllipsis(
+  ctx: CanvasRenderingContext2D,
+  text: string,
+  x: number,
+  maxWidth: number,
+  ellipsisString = '…',
```

Comment:
nit: our equivalent util function outside of canvas is called `truncate` with the ellipsisString instead named `suffix`. I prefer that nomenclature since it's more generic